eip-0029.md

```diff
@@ -0,0 +1,67 @@
+# Transaction purpose field
+
+* Author: MrStahlfelge
+* Status: Proposed
+* Created: 18-Feb-2022
+* License: CC0
+* Forking: not needed 
+
+## Motivation 
+
+Most payment systems provide a way to set a "purpose" field, used to display a short message
+from payment issuers to payment receivers.
+
+This EIP describes how one of the optional registers of Ergo boxes can be used to store a purpose
+message.
+This message can be set by wallet applications and dApps and can be shown by Ergo Explorer and
+Wallet applications.
+
+
+## Ergo boxes background
+
+A box can contain up to nine registers. Registers 1 to 3 are reserved and mandatory, registers 4
+to 9 are optional and can be used as input for scripts or are used for token minting (see EIP-4).
+
+Registers must be densely packed. It is not possible to use register 9 without adding register 4
+to 8.
+
+A box cannot be more than 4 kbytes.
+
+## Ergo purpose message standard
+
+Ergo boxes can hold Coll[Byte] that can be a UTF-8 representation of the purpose field.
+However, this would mean that we must designate a certain register to be used as purpose register,
+and because of the restrictions mentioned above this would mean to use R4, blocking it for all
+other use cases. This is not desired.
+
+To overcome this, we define that the purpose is encoded as follows in an arbitrary box:
+
+`Pair([0x50, 0x52, 0x50], Coll[Byte])`
+
+This is a pair of a constant Coll[Byte] used to identify this register as a purpose message
```

Comment:
How about "any" register as a purpose message? Any reason we can't *allow* people to put a memo running through all additional registers?